File: Documents/vsc/Java/gift1.java
package Java;
/*
ID: sumanm
LANG: JAVA
PROG: gift1
*/
import java.io.*;
import java.util.StringTokenizer;
public class gift1 {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new FileReader("gift1.in"));
        PrintWriter out = new PrintWriter(new BufferedWriter(new FileWriter("gift1.out")));
        String bNum = br.readLine();
        int num = Integer.parseInt(bNum);
        String[] people = new String[num];
        int[] balance = new int[num];
        for(int i = 0; i < num; i++){
            people[i] = br.readLine();
        }
        for(int i = 0; i < people.length; i++)
        {
            String temp = br.readLine();
            int index = -1;
            for(int j = 0; j < people.length; j++)
            {
                if(temp.equals(people[j]))
                {
                    index = j;
                    break;
                }
            }
            StringTokenizer st = new StringTokenizer(br.readLine());
            int money = Integer.parseInt(st.nextToken());
            int numP = Integer.parseInt(st.nextToken());
            int giveMoney = 0;
            if(numP != 0)
            {
                giveMoney = money/numP;
            }
            balance[index] -= giveMoney*numP;

            while(numP > 0)
            {
                String receiver = br.readLine();
                for(int j = 0; j < people.length; j++)
                {
                    if(receiver.equals(people[j]))
                    {
                        balance[j] += giveMoney;
                    }
                }
                numP--;
            }
        }
        for(int i = 0; i < balance.length; i++)
        {

            out.println(people[i]+" "+balance[i]);
        }
        br.close();
        out.close();
    }
}

File: Documents/vsc/C++/gift1.cpp
/*
ID: sumanm
LANG: C++
TASK: gift1
*/
#include <iostream>
#include <fstream>
#include <string.h>
using namespace std;

int num;
string people[10];
int balance[10];
    
int lookup(string name)
{
    for(int i = 0; i < num; i++) {
        if(name==people[i]) {
            return i;
        }
    }
    return 0;
}
int main(){
	ofstream fout ("gift1.out");
    ifstream fin ("gift1.in");
    int np;
    fin>>np;
    for(int i = 0; i < np; ++i)
    {
        fin>>people[i];
        num++;
    }
    for(int i = 0; i < num; ++i)
    {
        string name;
        int money, numP;
        fin>>name>>money>>numP;
        int index = lookup(name);
        for(int j = 0; j < numP; j++)
        {
            string receiver;
            fin>>receiver;
            int rIndex = lookup(receiver);
            balance[index] -= money/numP;
            balance[rIndex] += money/numP;
        }
    }
    for(int i = 0; i < np; i++)
    {
        fout<<people[i]<<" "<<balance[i]<<endl;
    }
    exit(0);
}
